manifests/repos.pp

```diff
@@ -22,19 +25,38 @@
       $package_key   = $docker::package_key
       $package_repos = $docker::package_repos
 
+      if ( $facts['os']['distro']['id'] == 'Debian' and versioncmp($facts['os']['distro']['release']['major'],'10') >= 0 ) or ( $facts['os']['distro']['id'] == 'Ubuntu' and versioncmp($facts['os']['distro']['release']['major'],'22') >= 0 ) { # lint:ignore:140chars
+        # fix deprecated apt-key warnings
+        file { $keyring:
+          ensure => file,
+          source => "puppet:///modules/${module_name}/etc/apt/keyrings/docker.gpg",
+          mode   => '0644',
+          owner  => 'root',
+          group  => 'root',
```

Comment:
To avoid more HTTP connections to the puppet server:

```suggestion
          ensure  => file,
          content => binary_file("${module_name}/etc/apt/keyrings/docker.gpg"),
          mode    => '0644',
          owner   => 'root',
          group   => 'root',
```